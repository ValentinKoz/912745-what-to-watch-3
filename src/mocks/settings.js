const genres = [`All genres`, `Adventure`, `Action`, `Comdey`, `Dramas`];

export {genres};
