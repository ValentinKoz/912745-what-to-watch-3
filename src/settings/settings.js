const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

const Path = {
  LOGIN: `/login`,
  FILMS: `/films`,
  MY_LIST: `/myList`,
  MAIN: `/`,
  REVIEW: `/review`,
  PLAYER: `/player`,
  PROMO: `/promo`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`,
};

const Namespace = {DATA: `DATA`, STATE: `STATE`, USER: `USER`};

const DISPLAYED_ITEMS = 8;

const ADD_TO_SHOW = 8;

export {MONTHS, ADD_TO_SHOW, DISPLAYED_ITEMS, Namespace, Path};

export const adaptedObject = (item) => {
  return {
    id: `${item.id}`,
    name: item.name,
    poster: item.poster_image,
    preview: item.preview_image,
    backgroundImg: item.background_image,
    backgroundColor: item.background_color,
    video: item.video_link,
    previewVideo: item.preview_video_link,
    description: item.description,
    rating: item.rating,
    scoresCount: item.scores_count,
    director: item.director,
    starring: item.starring,
    runTime: item.run_time,
    genre: item.genre,
    released: item.released,
    isFavorite: item.is_favorite
  };
};

export const adaptedAuth = (item) => {
  return {
    id: `${item.id}`,
    email: item.email,
    name: item.name,
    avatarUrl: item[`avatar_url`],
  };
};

export const getDate = (propDate) => {
  const date = new Date(propDate);
  const month = MONTHS[date.getMonth()];
  const days = date.getDay();
  const year = date.getFullYear();
  return `${month} ${days}, ${year}`;
};

export const getRatingLevel = (rating) => {
  if (rating >= 0 && rating < 3) {
    return `Bad`;
  } else if (rating >= 3 && rating < 5) {
    return `Normal`;
  } else if (rating >= 5 && rating < 8) {
    return `Good`;
  } else if (rating >= 8 && rating < 10) {
    return `Vary good`;
  } else if (rating === 10) {
    return `Awesome`;
  }
  return ``;
};
