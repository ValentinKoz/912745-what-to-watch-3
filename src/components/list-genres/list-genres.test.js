import React from "react";
import renderer from "react-test-renderer";
import ListGenres from "./list-genres.jsx";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Namespace} from "./../../mocks/settings.js";

const mockStore = configureStore([]);

const displayedItems = 8;

const films = [
  {
    id: `1`,
    name: `Name`,
    poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Snatch.jpg`,
    preview: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg`,
    backgroundImg: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Snatch.jpg`,
    backgroundColor: `#FDFDFC`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    description: `text...`,
    rating: 8.0,
    scoresCount: 43241,
    director: `Guy Ritchie`,
    starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
    runTime: 104,
    genre: `Comedy`,
    released: 2000,
    isFavorite: false
  },
  {
    id: `1`,
    name: `Name`,
    poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Snatch.jpg`,
    preview: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg`,
    backgroundImg: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Snatch.jpg`,
    backgroundColor: `#FDFDFC`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    description: `text...`,
    rating: 8.0,
    scoresCount: 43241,
    director: `Guy Ritchie`,
    starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
    runTime: 104,
    genre: `Comedy`,
    released: 2000,
    isFavorite: false
  }
];

it(`List Genres renderer correctly`, () => {
  const store = mockStore({
    [Namespace.DATA]: {
      films,
    },
    [Namespace.STATE]: {
      genre: `All genres`,
      displayedItems,
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <ListGenres
            onCardClickHandle={() => {}}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
