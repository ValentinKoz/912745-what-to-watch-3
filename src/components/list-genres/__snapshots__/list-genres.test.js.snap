// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Genres renderer correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        href=""
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href=""
        onClick={[Function]}
      >
        Comedy
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg"
          src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#"
          onClick={[Function]}
        >
          Name
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height="175"
          poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg"
          src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="#"
          onClick={[Function]}
        >
          Name
        </a>
      </h3>
    </article>
     
  </div>
  <div
    className="catalog__more"
    hidden={true}
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;
